com.unity.render-pipelines.high-definition/Runtime/Material/BuiltinGIUtilities.hlsl

```diff
@@ -43,15 +43,27 @@ void EvaluateLightmap(float3 positionRWS, float3 normalWS, float3 backNormalWS,
 #endif
 
 #ifdef LIGHTMAP_ON
-#ifdef DIRLIGHTMAP_COMBINED
-    SampleDirectionalLightmap(TEXTURE2D_ARGS(unity_Lightmap, samplerunity_Lightmap),
-        TEXTURE2D_ARGS(unity_LightmapInd, samplerunity_Lightmap),
-        uvStaticLightmap, unity_LightmapST, normalWS, backNormalWS, useRGBMLightmap, decodeInstructions, bakeDiffuseLighting, backBakeDiffuseLighting);
-#else
-    float3 illuminance = SampleSingleLightmap(TEXTURE2D_ARGS(unity_Lightmap, samplerunity_Lightmap), uvStaticLightmap, unity_LightmapST, useRGBMLightmap, decodeInstructions);
-    bakeDiffuseLighting += illuminance;
-    backBakeDiffuseLighting += illuminance;
-#endif
+    #ifdef DIRLIGHTMAP_COMBINED
+        #ifdef UNITY_DOTS_INSTANCING_ENABLED
+            SampleDirectionalLightmap(TEXTURE2D_ARRAY_ARGS(unity_Lightmaps, samplerunity_Lightmaps),
```

Comment:
hmm... thinking more about this and related to the code above, tell me if I am wrong but I guess we can't have both UNITY_DOTS_INSTANCING_ENABLED and non instancing enabled. In this case I will suggest to simply define a macro somewhere like this:

#if UNITY_DOTS_INSTANCING_ENABLED
#define TEXTURE2D_LIGHTMAP_ARGS  TEXTURE2D_ARRAY_ARGS
#define SAMPLE_TEXTURE2D_LIGHTMAP SAMPLE_TEXTURE2D_ARRAY
#define LIGHTMAP_EXTRA_ARGS , float slice
...
#else
#define TEXTURE2D_LIGHTMAP_ARGS  TEXTURE2D_ARGS
#define SAMPLE_TEXTURE2D_LIGHTMAP SAMPLE_TEXTURE2D
#define LIGHTMAP_EXTRA_ARGS
...
#endif

And thus no need to duplicate all the functions (and no need for the trick for include files).

Thought?

Reply:
Good idea. Will do.